webui/react/src/components/TaskList.tsx

```diff
@@ -530,24 +533,14 @@ const TaskList: React.FC<Props> = ({ workspace }: Props) => {
     }
   }, [fetchTasks, selectedTasks, updateSettings, canModifyWorkspaceNSC]);
 
-  const showConfirmation = useCallback(() => {
-    modal.confirm({
-      content: `
-        Are you sure you want to kill
-        all the eligible selected tasks?
-      `,
-      icon: <ExclamationCircleOutlined />,
-      okText: 'Kill',
-      onOk: handleBatchKill,
-      title: 'Confirm Batch Kill',
-    });
-  }, [handleBatchKill]);
-
   const handleBatchAction = useCallback(
     (action?: string) => {
-      if (action === Action.Kill) showConfirmation();
+      if (action === Action.Kill) {
+        setBatchAction(action);
```

Comment:
nit: since this is the only place `batchAction` is being set, you could remove the state and just pass `Action.Kill` to `BatchActionConfirmModal.Component` directly